### DISCUSSION TITLE
Determine if first row of pixels is fully transparent

### DISCUSSION THREAD

**Discussion by bullmoose20** [ImageMagick/ImageMagick#5494] [Help] [answered]
I was trying to figure out the best way to determine if a directory of 5000 transparent head shots contains "head chops" as shown in the attached image. So my thought was to use ImageMagick to determine if the first row of pixels is fully transparent... if yes, then the image is not a "head chop". If not, then I should get a report of those images and find a better head shot of the people in question.

Head cop:
![Adil El Arbi](https://user-images.githubusercontent.com/12549033/186913685-0e863c17-57ac-40f4-9ff9-c0142b34a42b.png)

Here is an example of an image that in theory should return that the first row is fully transparent:
![Aaron Taylor-Johnson](https://user-images.githubusercontent.com/12549033/186915148-40e82a57-12b4-4794-91b6-35fd52a9016e.png)

I have looked through this link and it seems promising, however, not quite what I was looking for:
https://legacy.imagemagick.org/discourse-server/viewtopic.php?t=36219#:~:text=To%20find%20the%20first%20fully%20transparent%20pixel%20in,transparency%20is%20represented%20as%20white%20rather%20than%20black.

**Comment by bullmoose20** [comment]
maybe this could work... but not sure if there is a better way or if this is even the proper way:
`PS C:\Users\Downloads\@people\test> magick '.\Adil El Arbi.png' -format "%@" info:`
`2000x3001+0+0`
`PS C:\Users\Downloads\@people\test> magick '.\Aaron Taylor-Johnson.png' -format "%@" info:`
`2000x2887+0+113`
If that last number is greater than 0, then that means that the first row is transparent... thoughts?

**Comment by snibgo** [comment, marked as answer]
"-trim" is quite complex and expensive. As you only care about the first row, first crop to that row. Then I suggest extracting the alpha channel. If the mean of that is zero, the row is entirely transparent. If the value is 1, it is entirely opaque.
```
magick head1.png -crop x1+0+0 +repage -alpha extract -format %[fx:mean] info:
0.324784

magick head2.png -crop x1+0+0 +repage -alpha extract -format %[fx:mean] info:
0
```

**Reply by bullmoose20** [reply]
Interesting.... because the above will let me incorporate the result and use a tolerance level.... because I may want to accept something that is greater than 0.... some people have 1 hair follicle that breaks the plane of that first row making it not equal to 0... but more than acceptable in terms of ultimately what I am trying to accomplish. So I will need to see what nonzero value would be considered acceptable.

**Comment by snibgo** [comment]
The `%[fx:...]` expression can be complex. For example, this gives "1" if the mean is greater than 0.01, otherwise it gives "0". Windows CMD syntax.
```
magick head1.png -crop x1+0+0 +repage -alpha extract -format "%[fx:mean>0.01?1:0]" info:
```
Beware that ">" is a special character in many shells. In Windows CMD, putting the format in double-quotes fixes that problem.

**Comment by bullmoose20** [comment]
I found the threshold for my images was to have anything that is greater than 0.06 considered a "head chop" situation and this seems to be a good compromise.
